galaxy_utils/sequence/fastq.py

```diff
@@ -593,6 +595,20 @@ def _close_on_error(self):
         # Extension point for subclasses (fastqVerboseErrorReader needs file access after error)
         return self.close()
 
+
+class fastqReader(Iterator, fileHandler):
+
+    mode = 'r'
+
+    def __init__(
+            self, fh=None, format='sanger', apply_galaxy_conventions=False, path=None, fix_id=False):
+        self.fh = fh
+        self.format = format
+        self.apply_galaxy_conventions = apply_galaxy_conventions
+        self.path = path
+        self.fix_id = fix_id  # fix inconsistent identifiers (source: SRA data dumps)
+        self._file = None
```

Comment:
You could move the initialisation of `fh`, `format`, `path` and `_file` attributes to `fileHandler.__init__()`

Reply:
thanks, I've done this in https://github.com/galaxyproject/sequence_utils/pull/27/commits/1e60240f263d67960f490be488b080f92404b5e1